src/app-logic.js

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useMemo } from 'react'
+import { useHistory } from 'react-router-dom'
+import { createAppHook, useApp } from '@aragon/connect-react'
+import connectVoting from '@aragon/connect-voting-disputable'
+
+function useSelectedProposal() {
+  const history = useHistory()
+
+  // TODO: get data from selected proposal
+  const selectedProposal = useMemo(() => {
+    return 'Vote information'
+  }, [])
+
+  const selectProposal = useCallback(
+    (proposalId) => {
+      history.push(`/proposals/${proposalId}`)
+    },
+    [history]
+  )
+  return [selectedProposal, selectProposal]
+}
+
+function useGetVotes() {
+  const useDisputableVoting = createAppHook(connectVoting)
+  const [disputableVoting] = useApp('disputable-voting')
+  const [votes] = useDisputableVoting(disputableVoting, (app) => app.votes())
+
+  return votes
+}
+
+// Handles the main logic of the app.
+export function useAppLogic() {
+  const [selectedProposal, selectProposal] = useSelectedProposal()
+  return {
+    selectProposal,
+    selectedProposal,
+    getVotes: useGetVotes(),
```

Comment:
Wasn't sure, but does `useGetVotes()` return a function? It looks like it just gets the votes, so maybe we should call this `votes`?

Was also curious for the future if we could get the loading state; does `useDisputableVoting()` provide that to us?

Reply:
I think in the future we can get the loading state with appsStatus.loading from useApps()

Reply:
I will take care of this in the next PR